### PR TITLE
Uplift theia dependencies to 1.34.2; document steps in CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by marco-miller**
This PR includes these commits, in order of appearance (from base to head):

1. **Align** `./package.json` `@theia/cli` dependency version.
2. **Fix** the `good-first-issues` label in `CONTRIBUTING.md`.
3. Add uplifting the Theia version to **CONTRIBUTING.md**.
4. **Uplift** theia dependencies from v`1.34.1` to `1.34.2`.
5. Add uplifting the **React** version in `CONTRIBUTING.md`.
6. Add uplifting the **Node** version in `CONTRIBUTING.md`.

Signed-off-by: Marco Miller <marco.miller@ericsson.com>

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -96,6 +96,26 @@ you'll learn high-level context about the project's different components without
   the code as well as learning the submission and review process (which is a significant learning step if you've
   never submitted a PR before).
 
+## Uplifting the Theia dependencies version
+
+Should one be willing to contribute such an uplift, these are the steps to consider, while
+more may be necessary depending on the case. Here is an example used to uplift from version
+`1.34.1` to `1.34.2`:
+
+1. `cd theia` or go to the Theia repository using a local terminal.
```
Comment (hoangphamEclipse) [inline CONTRIBUTING.md:105]: Can you make it 2, 3, 4, and so on since it is an ordered list?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -115,6 +115,12 @@ more may be necessary depending on the case. Here is an example used to uplift f
    * `./examples/electron/package.json`
    * `./package.json`
    * `./theia-extensions/viewer-prototype/package.json`
+1. Align these React dependency versions in `./package.json` with Theia's `packages/core/package.json`, when applicable:
```
Comment (hoangphamEclipse) [inline CONTRIBUTING.md:118]: Same here, can you continue the numbering?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -121,6 +121,8 @@ more may be necessary depending on the case. Here is an example used to uplift f
    * `"@types/react"`
    * `"@types/react-dom"`
 1. Do the same also for `react-test-renderer` in `packages/react-components/package.json`, and potentially elsewhere in the extension.
+1. Align `node-version` in `.github/workflows` yaml files with Theia's.
```
Comment (hoangphamEclipse) [inline CONTRIBUTING.md:124]: Same here

**Review by hoangphamEclipse — CHANGES_REQUESTED** [review]
Thank you for the patch Marco, just a small comment regarding the numbering of the uplifting steps in the documentation.

File: CONTRIBUTING.md (hunk 1 above)
Comment (marco-miller) [reply CONTRIBUTING.md:105]: This shouldn't be necessary in Markdown.
- Using `1.` across leads to [automated numbering (increments) at rendering time](https://github.com/eclipse-cdt-cloud/theia-trace-extension/blob/1d0684e4113ada3ac3cfaaf7894c07a8938d65f0/CONTRIBUTING.md).
- It also allows for not having to manually maintain the said numbering (increments).

File: CONTRIBUTING.md (hunk 2 above)
Comment (marco-miller) [reply CONTRIBUTING.md:118]: -Per above. Thanks for double-checking from your end.

File: CONTRIBUTING.md (hunk 3 above)
Comment (marco-miller) [reply CONTRIBUTING.md:124]: (Per further above.)

**Review by hoangphamEclipse — APPROVED** [review]

**Pull request merged**